### PR TITLE
p2p: avoid spinning loop on out-of-handles

### PR DESCRIPTION
**Pull request by holiman**
Related to #https://github.com/ethereum/go-ethereum/issues/21877 .

In the event that we've (temporarily) run out of file handles, the p2p server will spinloop on trying to `Accept`. This PR adds a little sleep, to not spin quite so fast, and also to consider the "temporary" error permanent after 2 seconds. 

Without this fix, `12M` log entries about temporary read error: 
```
archive@archive:~$ cat dump2.log | grep "Temporary read error" | wc -l
12832728
```

### CODE REVIEW

File: p2p/server.go [hunk 1]
```diff
@@ -854,13 +854,16 @@ func (srv *Server) listenLoop() {
...
-			fd  net.Conn
-			err error
+			fd        net.Conn
+			err       error
+			tmpErrors = uint64(0)
 		)
 		for {
 			fd, err = srv.listener.Accept()
-			if netutil.IsTemporaryError(err) {
+			if netutil.IsTemporaryError(err) && tmpErrors < 200 {
```
Comment (fjl) [inline p2p/server.go:863]: I'm not sure I like this limit. The delay is OK, but the limit can be a problem: if the node runs out of file descriptors temporarily (i.e. not during shutdown), listening will not recover once it's over the threshold.

Comment (fjl) [reply p2p/server.go:863]: The solution could be to simply make the delay longer and remove the limit.

Comment (holiman) [reply p2p/server.go:863]: Ok, what's your referred delay? How about starting at `10ms`, then adding doubling the delay but cap the delay at 2 seconds  ?

Comment (fjl) [reply p2p/server.go:863]: I think 2s is too much. It might be OK to just use a 200ms static delay here.

Comment (holiman) [reply p2p/server.go:863]: I agree, delay-wise, however, spitting out 5 lines of logs per second seems excessive, so we should still limit the log output to max one line per second, IMO

Comment (holiman) [reply p2p/server.go:863]: Ok, fixed, PTAL

**Review by fjl — APPROVED** [review]

**Pull request merged**